src/frontend/Semantic_check.ml

```diff
@@ -679,8 +679,8 @@ and semantic_check_expression cf ({emeta; expr} : Ast.untyped_expression) :
       semantic_check_variable cf emeta.loc id
       |> Validate.apply_const (semantic_check_identifier id)
   | IntNumeral s -> (
-    match int_of_string_opt s with
-    | Some i when i < 2_147_483_648 ->
+    match float_of_string_opt s with
```

Comment:
`int_to_string` seems to not parse an integer literal from a string model. I have not figured why, but parsing and comparing it as a float works for both the binary and stancjs version.

Reply:
so it just produced `None`? IIRC Javascript doesn't have a native integer type, maybe that has something to do with it.

Reply:
Sorry, I meant the parsing does not work correctly.
It behaves as if 0 is not < max int. Example:
```ocaml
match int_of_string_opt s with
    | Some i when i < 2_147_483_648 ->
        mk_typed_expression ~expr:(IntNumeral s) ~ad_level:DataOnly ~type_:UInt
          ~loc:emeta.loc
        |> Validate.ok
    | Some _i when -> (* This gets used. If you print _i, you get 0. *)
        mk_typed_expression ~expr:(IntNumeral s) ~ad_level:DataOnly ~type_:UInt
          ~loc:emeta.loc
        |> Validate.ok
    | _ -> Semantic_error.bad_int_literal emeta.loc |> Validate.error )
```

Reply:
But yeah, JS integer support is probably the issue here. Though we also use int_of_string for other line and column numbers in errors and those seem to work fine.

Reply:
Huh, weird.